src/components/course/course-header/CourseImportantDates.jsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Col, Icon, Row, Stack,
+} from '@openedx/paragon';
+import { Calendar } from '@openedx/paragon/icons';
+import dayjs from 'dayjs';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { defineMessages, useIntl } from '@edx/frontend-platform/i18n';
+import { DATE_FORMAT } from '../data';
+import {
+  determineAllocatedCourseRunAssignmentsForCourse,
+  getSoonestEarliestPossibleExpirationData,
+  useCourseMetadata,
+  useRedeemablePolicies,
+} from '../../app/data';
+
+const messages = defineMessages({
+  importantDates: {
+    id: 'enterprise.course.about.page.important-dates.title',
+    defaultMessage: 'Important dates',
+    description: 'Title for the important dates section on the course about page',
+  },
+  enrollByDate: {
+    id: 'enterprise.course.about.page.important-dates.enroll-by-date',
+    defaultMessage: 'Enroll-by date',
+    description: 'Enroll-by date for the important dates section on the course about page',
+  },
+  courseStarts: {
+    id: 'enterprise.course.about.page.important-dates.course-starts',
+    defaultMessage: 'Course starts',
+    description: 'Course starts for the important dates section on the course about page in future tense',
+  },
+  courseStarted: {
+    id: 'enterprise.course.about.page.important-dates.course-started',
+    defaultMessage: 'Course started',
+    description: 'Course started the important dates section on the course about page in past tense',
+  },
+});
+
+const CourseImportantDates = () => {
+  const { courseKey } = useParams();
+  const { data: redeemableLearnerCreditPolicies } = useRedeemablePolicies();
+  const { data: courseMetadata } = useCourseMetadata();
+  const intl = useIntl();
+  const {
+    allocatedCourseRunAssignments,
+    allocatedCourseRunAssignmentKeys,
+    hasAssignedCourseRuns,
+  } = determineAllocatedCourseRunAssignmentsForCourse({
+    redeemableLearnerCreditPolicies,
+    courseKey,
+  });
+
+  const [searchParams] = useSearchParams();
+  const courseRunKey = searchParams.get('course_run_key')?.replaceAll(' ', '+');
+
+  if (!hasAssignedCourseRuns || (!!courseRunKey && !allocatedCourseRunAssignmentKeys.includes(courseRunKey))) {
+    return null;
+  }
+
+  const { date, soonestExpirationDateData } = getSoonestEarliestPossibleExpirationData({
+    assignmentObjectArray: allocatedCourseRunAssignments,
+    dateFormat: `${DATE_FORMAT} h:mm A`,
+  });
+  const soonestExpiringAllocatedAssignmentCourseStartDate = courseMetadata.availableCourseRuns.find(
+    (courseRun) => courseRun.key === soonestExpirationDateData?.contentKey,
+  )?.start;
+
+  const enrollByDate = date;
+  const courseStartDate = dayjs(soonestExpiringAllocatedAssignmentCourseStartDate).format(DATE_FORMAT);
+  const hasCourseStarted = dayjs(courseStartDate).isBefore(dayjs());
+
+  return (
+    <Stack gap={2} className="mt-4">
+      <h3>{intl.formatMessage(messages.importantDates)}</h3>
+      <section>
```

Comment:
[design] When testing this locally, it's consistent that on my screen resolution/size, the "Enroll-by date" row wraps onto a new line, which isn't super ideal. I wonder if we should consider rendering these dates with `small` typography to eliminate wrapping of the date format, which makes the design look "off" IMHO.

**Current**

![image](https://github.com/user-attachments/assets/86537e9c-81f2-4761-9128-7e13c81c7410)

**Proposed**

![image](https://github.com/user-attachments/assets/07e2ef46-d13b-4274-abee-bbf43ee7c390)

Note how there's less immediate attention drawn to the dates when they're smaller, where the learner's focus can remain more about the _content_'s relevance before getting drawn into considering dates.